survey/src/main/java/com.quickbirdstudios.surveykit/steps/QuestionStep.kt

```diff
@@ -15,6 +18,8 @@ class QuestionStep(
     val text: String,
     val nextButton: String = "Next",
     val answerFormat: AnswerFormat,
+    var lifecycle: Lifecycle? = null,
+    var addressProvider: AddressSuggestionProvider? = null,
```

Comment:
I'm not really a fan of this, since now you have the option to add a lifecycle here even if it's not necessary. How about this, we remove these 2 here, but add them here
```kotlin
    data class LocationAnswerFormat(
        val lifecycle: Lifecycle,
        val addressProvider: AddressSuggestionProvider? = null
    ) : AnswerFormat() {

        @Parcelize
        data class Location(val latitude: Double, val longitude: Double) : Parcelable {
            override fun toString(): String {
                return "${this.latitude}:${this.longitude}"
            }
        }
    }
```
then here we do:
```kotlin
    private fun createLocationPickerQuestion(
        context: Context, stepResult: StepResult?
    ): LocationPickerQuestionView {
        val castedAnswerFormat =  this.answerFormat as LocationAnswerFormat
        return LocationPickerQuestionView(
            context = context,
            id = id,
            title = title,
            text = text,
            isOptional = isOptional,
            nextButtonText = nextButton,
            lifecycle = castedAnswerFormat.lifecycle,
            addressProvider = castedAnswerFormat.addressProvider
                ?: GeocoderAddressSuggestionProvider(context),
            answerFormat = this.answerFormat,
            preselected = stepResult.toSpecificResult<LocationQuestionResult>()?.answer
        )
    }
```
and when you want to use it, it looks like this:
```kotlin
            QuestionStep(
                title = this.resources.getString(R.string.location_select_title),
                text = this.resources.getString(R.string.location_question_text),
                answerFormat = AnswerFormat.LocationAnswerFormat(
                    lifecycle = lifecycle,
                    addressProvider = GeocoderAddressSuggestionProvider(this)
                )
            ),
```


What do you think?

Reply:
I think this is more proper way, cause these fields only using on LocationAnswerFormat. I will implement like that.